eq-author-api/middleware/identification/index.js

```diff
@@ -0,0 +1,60 @@
+const verifyServiceRequest = require("./verifyServiceRequest");
+const { isNil, isEmpty } = require("lodash/fp");
+const jwt = require("jsonwebtoken");
+const { getUserByExternalId } = require("../../utils/datastore");
+
+module.exports = logger => async (req, res, next) => {
+  const authHeader = req.header(process.env.AUTH_HEADER_KEY || "authorization");
+  if (isNil(authHeader)) {
+    logger.error("Request must contain a valid authorization header.");
+    res.send(401);
+    return;
+  }
+
+  const accessToken = authHeader.replace("Bearer ", "").replace(/=/g, "");
+  if (isEmpty(accessToken)) {
+    logger.error("Request must contain a valid access token.");
+    res.send(401);
+    return;
+  }
+
+  const jwtToken = jwt.decode(accessToken);
+  if (isNil(jwtToken)) {
+    logger.error("Could not decode JWT token.");
+    res.send(401);
+    return;
+  }
+
+  if (jwtToken.serviceName) {
+    await verifyServiceRequest(accessToken, jwtToken.serviceName);
+    req.user = {
+      name: jwtToken.serviceName,
+      serviceVerified: true,
```

Comment:
Could use the same boolean as `userRegistered` if we come up with a better name

Reply:
I had them both as the same name at first, think it was `isValidIdentity` switched it to two seperate flags though as it felt better.

Reply:
It would be nice if the rest of the app didn't care where it came from. So it feels a little odd to leak the multiple types of identities.

Reply:
The app might care about where it comes from in future, while at the moment services and users are equal my preference would be to keep them separate and just be a little more verbose in the rejection middleware.

Reply:
I'd rather it be "pure" until we get the use case for it not being so later on